src/System.IO.FileSystem/src/System/IO/FileSystem.Unix.cs

```diff
@@ -98,6 +105,37 @@ public static void MoveFile(string sourceFullPath, string destFullPath)
                 return;
             }
 
+            // If overwrite is allowed then just call rename
+            if (overwrite)
+            {
+                if (Interop.Sys.Rename(sourceFullPath, destFullPath) < 0)
+                {
+                    Interop.ErrorInfo errorInfo = Interop.Sys.GetLastErrorInfo();
+                    if (errorInfo.Error == Interop.Error.EXDEV) // rename fails across devices / mount points
+                    {
+                        CopyFile(sourceFullPath, destFullPath, overwrite);
+                    }
+                    else if (errorInfo.Error == Interop.Error.ENOENT && !Directory.Exists(Path.GetDirectoryName(destFullPath))) // The parent directory of destFile can't be found
+                    {
+                        // Windows distinguishes between whether the directory or the file isn't found,
+                        // and throws a different exception in these cases.  We attempt to approximate that
+                        // here; there is a race condition here, where something could change between
+                        // when the error occurs and our checks, but it's the best we can do, and the
+                        // worst case in such a race condition (which could occur if the file system is
+                        // being manipulated concurrently with these checks) is that we throw a
+                        // FileNotFoundException instead of DirectoryNotFoundexception.
+                        throw Interop.GetExceptionForIoErrno(errorInfo, destFullPath, isDirectory: true);
+                    }
+                    else
+                    {
+                        throw Interop.GetExceptionForIoErrno(errorInfo);
+                    }
```

Comment:
Can these last two blocks just be consolidated to be:
```C#
else
{
    throw Interop.GetExceptionForIoErrno(errorInfo, destFullPath, isDirectory: errorInfo.Error == Interop.Error.ENOENT && !Directory.Exists(Path.GetDirectoryName(destFullPath)));
}
```
?

Reply:
Change made.